src/Home/useBluetoothStatus.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "react"
+import { Platform, AppState } from "react-native"
+import { isBluetoothEnabled } from "../gaen/nativeModule"
+
+export const useBluetoothStatus = (): boolean => {
+  const [btStatus, setBTStatus] = useState(false)
+  useEffect(() => {
+    const fetchBTEnabled = async () => {
+      const status = await fetchBTStatus()
+      setBTStatus(status)
+    }
+
+    AppState.addEventListener(getOSListener(), () => fetchBTEnabled())
+    return AppState.removeEventListener(getOSListener(), () => fetchBTEnabled())
```

Comment:
For my own learning, can you explain what this line does?

Reply:
Absolutely! A function returned from the `useEffect` hook is called before the component unmounts (it's similar to `componentWillUnmount` from the olden days), so it's often used to define what "clean up" actions you want to run before this component is removed from the tree. In this case, since we are adding an event listener, we we can remove the event listener so that it doesn't hang around and keep running these fetch requests when the user navigates away from the Home screen.

Reply:
Ohhhh I see that makes sense. Thank you! 🙏🏼